docs/index.md

```diff
@@ -0,0 +1,44 @@
+[![npm last version](https://img.shields.io/npm/v/nextcloud-vue.svg?style=flat-square)](https://www.npmjs.com/package/nextcloud-vue)
+[![travis build status](https://img.shields.io/travis/com/nextcloud/nextcloud-vue/master.svg?style=flat-square)](https://travis-ci.com/nextcloud/nextcloud-vue)
+[![Dependabot status](https://img.shields.io/badge/Dependabot-enabled-brightgreen.svg?longCache=true&style=flat-square&logo=dependabot)](https://dependabot.com)
+[![Codacy Badge](https://img.shields.io/codacy/grade/57e9764b68904cbf8f9e050c33340ab4.svg?style=flat-square)](https://app.codacy.com/app/skjnldsv/nextcloud-vue)
+[![Code coverage](https://img.shields.io/codecov/c/github/nextcloud/nextcloud-vue.svg?style=flat-square)](https://codecov.io/gh/nextcloud/nextcloud-vue/)
+[![irc](https://img.shields.io/badge/IRC-%23nextcloud--dev%20on%20freenode-blue.svg?style=flat-square)](https://webchat.freenode.net/?channels=nextcloud-dev)
+
+This repo contains the various Vue.js components that Nextcloud uses for its internal design and structure.
+
+# Installation
+
+```bash
+npm i --save nextcloud-vue
+```
+
+# Usage
+```js static
+import { AppNavigation } from 'nextcloud-vue'
+
+export default {
+    name: 'MyComponent',
+    components: {
+        AppNavigation
+    }
+}
+```
+
+You can also import individual module without bundling the full library.
+
+
+```js static
+import Avatar from 'nextcloud-vue/dist/Components/Avatar'
+```
+
+## Recommendations
+
+Some components require Nextcloud functionality to work, so it is currently recommended to extend Vue with the following:
+
+```js static
+Vue.prototype.t = window.t
+Vue.prototype.n = window.n
+Vue.prototype.OC = window.OC
+Vue.prototype.OCA = window.OCA
```

Comment:
For what is this needed?

If it's only for lint checks, extending [`plugin:nextcloud/recommended`](https://github.com/nextcloud/eslint-plugin-nextcloud) in eslint configuration should be better.

Reply:
Probably only t and OC are required, but they don't hurt as long as we don't ship translations with the components:
https://github.com/nextcloud/nextcloud-vue/search?l=Vue&q=t%28%27core%27
https://github.com/nextcloud/nextcloud-vue/search?q=getCurrentUser&unscoped_q=getCurrentUser

Reply:
Ah, OC probably also not needed since it is accessing the global object directly without the Vue object as a proxy.

Reply:
I don't like that app developers have to care about internals from the used library. We should add a [mixin like this](https://github.com/nextcloud/notes/blob/master/src/mixins/AppGlobal.js) to nc-vue and use it in those components which use `t` or similar. App developers then don't have to add this prototype code anymore. They can even use this mixin, too, when they want to us `t` in their app.

Reply:
Yes, sounds good. I just added it here as I encountered it in the docs build.

Reply:
Fine, I can do a PR for the mixin, later (in the next days, I think).